extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampStreamExtractor.java

```diff
@@ -0,0 +1,335 @@
+    }
+
+    @Nonnull
+    @Override
+    public Description getDescription() {
+        String s = BandcampExtractorHelper.smartConcatenate(
+                new String[]{
+                        current.getString("about"),
+                        current.getString("lyrics"),
+                        current.getString("credits")
+                }, "\n\n"
+        );
+        return new Description(s, Description.PLAIN_TEXT);
+    }
+
+    @Override
+    public int getAgeLimit() {
+        return NO_AGE_LIMIT;
+    }
+
+    @Override
+    public long getLength() {
+        return 0;
+    }
+
+    @Override
+    public long getTimeStamp() {
+        return 0;
+    }
+
+    @Override
+    public long getViewCount() {
+        return -1;
+    }
+
+    @Override
+    public long getLikeCount() {
+        return -1;
+    }
+
+    @Override
+    public long getDislikeCount() {
+        return -1;
+    }
+
+    @Nonnull
+    @Override
+    public String getDashMpdUrl() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public String getHlsUrl() {
+        return null;
+    }
+
+    @Override
+    public List<AudioStream> getAudioStreams() {
+        List<AudioStream> audioStreams = new ArrayList<>();
+
+        audioStreams.add(new AudioStream(
+                albumJson.getArray("trackinfo").getObject(0)
+                        .getObject("file").getString("mp3-128"),
+                MediaFormat.MP3, 128
+        ));
+        return audioStreams;
+    }
+
+    @Override
+    public List<VideoStream> getVideoStreams() {
+        return null;
+    }
+
+    @Override
+    public List<VideoStream> getVideoOnlyStreams() {
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public List<SubtitlesStream> getSubtitlesDefault() {
+        return new ArrayList<>();
+    }
+
+    @Nonnull
+    @Override
+    public List<SubtitlesStream> getSubtitles(MediaFormat format) {
+        return new ArrayList<>();
+    }
+
+    @Override
+    public StreamType getStreamType() {
+        return StreamType.AUDIO_STREAM;
+    }
+
+    @Override
+    public StreamInfoItem getNextStream() {
+        return null;
+    }
+
+    @Override
+    public StreamInfoItemsCollector getRelatedStreams() {
+        return null;
+    }
+
+    @Override
+    public String getErrorMessage() {
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getHost() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public String getPrivacy() {
+        return "";
+    }
+
+    @Nonnull
+    @Override
+    public String getCategory() {
+        // Get first tag from html, which is the artist's Genre
+        return document.getElementsByAttributeValue("itemprop", "keywords").first().text();
+    }
+
+    @Nonnull
+    @Override
+    public String getLicence() {
+
+        int license = current.getInt("license_type");
+
+        // Tests resulted in this mapping of ints to licence: https://cloud.disroot.org/s/ZTWBxbQ9fKRmRWJ/preview
```

Comment:
where did you find this screenshot (I don't mean with the numbers at the left, but the original one).
You don't know about 7? it's weird they have jumped one

Reply:
It's a screenshot you can take with a bandcamp artist account when opening the preferences of a track. These are the only options you can select, I don't know a way to get 7.